polar2grid/writers/hdf5.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import annotations
+from typing import TextIO
+
+import os
+import sys
+import logging
+
+import satpy
+import h5py
+import numpy as np
+import xarray as xr
+import dask.array as da
+
+from itertools import groupby
+from datetime import datetime as datetime
+from satpy.dataset import DataID
+from satpy.writers import ImageWriter
+
+from trollsift import Parser
+
+from pyresample.geometry import SwathDefinition
+from polar2grid.utils.legacy_compat import convert_p2g_pattern_to_satpy
+from polar2grid.writers.geotiff import NUMPY_DTYPE_STRS, NumpyDtypeList, str_to_dtype, int_or_float
+
+from polar2grid.core.script_utils import setup_logging, rename_log_file, create_exc_handler
+
+USE_POLAR2GRID_DEFAULTS = bool(int(os.environ.setdefault("USE_POLAR2GRID_DEFAULTS", "1")))
+
+LOG = logging.getLogger(__name__)
+
+# reader_name -> filename
+DEFAULT_OUTPUT_FILENAMES = {None: "{platform_name}_{sensor}_{start_time:%Y%m%d_%H%M%S}.h5"}
+
+
+def all_equal(iterable: list[str]) -> bool:
+    "Returns True if all the elements are equal to each other"
```

Comment:
Three quotes and a period at the end. I'm surprised black doesn't complain about/fix this.